stable/rabbitmq/templates/statefulset.yaml

```diff
@@ -122,12 +122,29 @@ spec:
             valueFrom:
               fieldRef:
                 fieldPath: status.podIP
-          - name: RABBITMQ_USE_LONGNAME
-            value: "true"
+          - name: MY_POD_NAME
+            valueFrom:
+              fieldRef:
+                fieldPath: metadata.name
+          - name: MY_POD_NAMESPACE
+            valueFrom:
+              fieldRef:
+                fieldPath: metadata.namespace
+          - name: K8S_SERVICE_NAME
+            value: "{{ template "rabbitmq.fullname" . }}-headless"
+          - name: K8S_ADDRESS_TYPE
+            value: {{ default "ip" .Values.rabbitmq.clustering.address_type }}
+          {{- if (eq "hostname" .Values.rabbitmq.clustering.address_type) }}
+          - name: RABBITMQ_NODENAME
+            value: "rabbit@$(MY_POD_NAME).$(K8S_SERVICE_NAME).$(MY_POD_NAMESPACE).svc.cluster.local"
```

Comment:
I would also parametrize the `cluster.local` domain in case someone has a cluster domain different to the default one